### PR TITLE
Bluetooth: controller: Implement LE Set Host Feature Command

### PR DESCRIPTION
**Pull request by cvinayak**
Replace the use of LL_FEAT define with ll_feat_get() so that
feature set value can be updated at runtime with host
feature bit values.

Added implementation of HCI LE Set Host Feature command.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ll_feat.c [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
-	ARG_UNUSED(bit_number);
-	ARG_UNUSED(bit_value);
+	host_features &= ~BIT64(bit_number);
+	if (bit_value) {
+		host_features |= BIT64(bit_number);
+	}
 
-	return BT_HCI_ERR_CMD_DISALLOWED;
+	return BT_HCI_ERR_SUCCESS;
+}
```
Comment (thoh-ot) [inline subsys/bluetooth/controller/ll_sw/ll_feat.c:28]: Per BLUETOOTH CORE SPECIFICATION Version 5.2 | Vol 4, Part E, 7.8.115 LE Set Host Feature Command:

> If Bit_Number specifies a feature bit that is not controlled by the Host, the
> Controller shall return the error code Unsupported Feature or Parameter Value
> (0x11).

> If the Host issues this command while the Controller has an established ACL,
> the Controller shall return the error code Command Disallowed (0x0C).

**Review by thoh-ot — CHANGES_REQUESTED** [review]

**Review by thoh-ot — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_conn.c [hunk 2]
```diff
@@ -3122,9 +3127,6 @@ static inline void event_fex_prep(struct ll_conn *conn)
 		/* procedure request acked, move to waiting state */
 		conn->llcp_feature.ack--;
 
-		/* use initial feature bitmap */
-		conn->llcp_feature.features_conn = LL_FEAT;
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c:3126]: As far as I can tell, there are some code paths in the slave role that can reach this point without having `features_conn` initialized. Are you sure this line is not needed?

**Comment by wopu-ot** [comment]
The summary of the first commit is somewhat misleading. The commit replaces a macro with a function rather than adding support for 64-bit features bitmap.

File: subsys/bluetooth/controller/ll_sw/ull_conn.c (hunk 2 above)
Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c:3126]: Isn't the conn object is being setup in ll_adv_enable() ?

Comment (wopu-ot) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c:3126]: You are right. Thanks for pointing that out.

**Review by wopu-ot — CHANGES_REQUESTED** [review]
The commit message of the first commit should be updated to match the content of the commit, otherwise LGTM.

File: subsys/bluetooth/controller/ll_sw/ll_feat.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ll_feat.c]: done

**Comment by cvinayak** [comment]
> 
> 
> The commit message of the first commit should be updated to match the content of the commit, otherwise LGTM.

@wopu-ot Done.

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**